### PR TITLE
migrated from RabbitMQ to Kafka (GDEV-1325)

### PR DESCRIPTION
**Pull request by KerstenBreuer**
```
and update to current event schemas.

Bumps version to 0.4.0.

Co-authored-by: Moritz Hahn <75744178+MoritzHahn1337@users.noreply.github.com>
Co-authored-by: Thomas Zajac <thomas-jakob.zajac@uni-tuebingen.de>
```

https://editor.swagger.io/?url=https://raw.githubusercontent.com/ghga-de/download-controller-service/c780382c6d9719896774ffd6e06343a8a7b8b453/openapi.yaml

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -53,7 +53,6 @@ console_scripts =
 dev =
     ghga-service-chassis-lib[dev]==0.16.1
```
Comment (MoritzHahn1337) [inline setup.cfg:54]: `chassis lib[dev]` could probably be deleted now.

**Review by MoritzHahn1337 — DISMISSED** [review]
Just some typos again :-D

File: dcs/container.py [hunk 2]
```diff
@@ -32,21 +34,34 @@ class Container(ContainerBase):
...
 
     # outbound translators:
     drs_object_dao = get_constructor(DrsObjectDaoConstructor, dao_factory=dao_factory)
+    event_publisher = get_constructor(
+        EventPubTranslator, config=config, provider=event_pub_provider
+    )
 
-    # outbound adapters:
-
+    # outbound adapters (not following the triple hexagonal translator/provider
```
Comment (MoritzHahn1337) [inline dcs/container.py:45]:
```suggestion
    # outbound adapters (not following the triple hexagonal translator/provider)
```

We should think about implementing a static code analys tool that looks for unclosed parenthesis in docs :-D

File: dcs/adapters/outbound/event_pub.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    def __init__(
+        self, *, config: EventPubTranslatorConfig, provider: EventPublisherProtocol
+    ):
+        """Initialize with configs and a provider of the EventPublisherProtocol."""
+
+        self._config = config
+        self._provider = provider
+
+    async def download_served(self, *, drs_object: models.DrsObjectWithUri) -> None:
+        """Communicate the event of an download being served. This can be relevant for
```
Comment (mephenor) [inline dcs/adapters/outbound/event_pub.py:88]:
```suggestion
        """Communicate the event of a download being served. This can be relevant for
```

**Review by mephenor — APPROVED** [review]

File: setup.cfg (hunk 1 above)
Comment (KerstenBreuer) [reply setup.cfg:54]: I think we would still need it as this is defining all the dev dependencies for mypy and co.

**Review by MoritzHahn1337 — APPROVED** [review]

**Review by mephenor — APPROVED** [review]

**Pull request merged**